providers/azure.go

```diff
@@ -224,14 +227,26 @@ func (p *AzureProvider) prepareRedeem(redirectURL, code, codeVerifier string) (u
 	if code == "" {
 		return params, ErrMissingCode
 	}
-	clientSecret, err := p.GetClientSecret()
-	if err != nil {
-		return params, err
+
+	if !p.useWorkloadIdentity {
+		clientSecret, err := p.GetClientSecret()
+		if err != nil {
+			return params, err
+		}
+		params.Add("client_secret", clientSecret)
+	} else {
+		federatedTokenPath := os.Getenv("AZURE_FEDERATED_TOKEN_FILE")
+		federatedToken, err := os.ReadFile(federatedTokenPath)
```

Comment:
Do we need to read the file every time the app processes the authentication of a user? How often is this expected to change?

Reply:
Yes, federated credentials are supposed to be rather short-lived and updates in place. They are injected into the pod via Admission control.